api/v1alpha3/cloudprovider_encoding.go

```diff
@@ -250,8 +250,12 @@ func isEmpty(val reflect.Value) bool {
 // MarshalCloudProviderArgs marshals the cloud provider arguments for passing
 // into a pod spec
 func (c *CPIConfig) MarshalCloudProviderArgs() []string {
-	c.ProviderConfig.Cloud.ExtraArgs = map[string]string{}
-	args := c.ProviderConfig.Cloud.ExtraArgs
+	var args map[string]string
+	if c.ProviderConfig.Cloud != nil {
+		args = c.ProviderConfig.Cloud.ExtraArgs
+	} else {
+		args = make(map[string]string, 3)
+	}
 	args["--v"] = "2"
```

Comment:
If `c.ProviderConfig.Cloud == nil` I'd return an empty map, as we won't deploy the ccm